maven-model-builder/src/main/java/org/apache/maven/model/io/DefaultModelReader.java

```diff
@@ -48,14 +54,44 @@
 public class DefaultModelReader
     implements ModelReader
 {
+    @Inject
+    private ModelSourceTransformer transformer;
 
+    public void setTransformer( ModelSourceTransformer transformer )
+    {
+        this.transformer = transformer;
+    }
+    
     @Override
     public Model read( File input, Map<String, ?> options )
         throws IOException
     {
         Objects.requireNonNull( input, "input cannot be null" );
 
-        Model model = read( new FileInputStream( input ), options );
+        TransformerContext context = null;
+        if ( options != null )
+        {
+            context = (TransformerContext) options.get( "transformerContext" );
+        }        
+
+        final InputStream is;
+        if ( context == null )
+        {
+            is = new FileInputStream( input );
+        }
+        else
+        {
+            try
+            {
+                is = transformer.transform( input.toPath(), context );
```

Comment:
You have opened the input streams, but never closed them.

Reply:
fixed